skimage/util/lazy.py

```diff
@@ -0,0 +1,65 @@
+import sys
+import importlib
+import importlib.util
+
+
+def install_lazy(module_name, submodules=None, submod_funcs=None):
+    """Install lazily loaded submodules and functions.
+
+    Parameters
+    ----------
+    module_name : str
+        Typically use __name__.
+    submodules : list
+        List of submodules to install.
+    submod_funcs : dict
+        Dictionary of submodule -> list of functions.  These functions are
+        imported as they are used.
+    """
+    if submod_funcs is None:
+        submod_funcs = {}
+
+    if submodules is None:
+        submodules = []
+
+    all_funcs = []
+    for mod, funcs in submod_funcs.items():
+        all_funcs.extend(funcs)
+
+    def __getattr__(name):
+        if name in submodules:
+            lazy_mod = require(f'skimage.{name}')
```

Comment:
I really don't like how the name of the package is hard coded here. I'm not sure if there is an easy way around it, but I would prefer an implementation of this utility that could easily be plucked out of skimage and moved to importlib itself (or another third-party utility library).  I'm not sure what the best way to do that is ATM, but I'm just throwing that out there.

Reply:
Fixed